src/main/java/org/spongepowered/api/item/inventory/Container.java

```diff
@@ -24,60 +24,50 @@
  */
 package org.spongepowered.api.item.inventory;
 
-import org.spongepowered.api.entity.living.player.Player;
-import org.spongepowered.api.item.inventory.type.Interactable;
-import org.spongepowered.api.plugin.PluginContainer;
-
-import java.util.Set;
+import java.util.List;
+import java.util.Optional;
 
 /**
  * A Container is effectively a <em>ViewModel</em> for a particular set of
  * {@link Inventory} objects used to allow players to interact
  * with the Inventories, usually via a GUI (the View).
  */
-public interface Container extends Interactable {
+public interface Container extends Inventory {
 
-    /**
-     * Gets the current viewers looking at this Inventory.
+     /**
+     * Returns whether given slot is part of the viewed inventories
+     * but not part of the viewers own inventory.
      *
-     * @return The current viewers of this inventory
-     */
-    Set<Player> getViewers();
-
-    /**
-     * Checks for whether this Inventory currently has viewers.
+     * <p>Examples for viewed inventory slots:</p>
+     * <p>All the slots of the opened chest and not the player grid below.</p>
+     * <p>The slots of the crafting grid in the opened player inventory.</p>
      *
-     * @return True if viewers are currently looking at this inventory
+     * @param slot The slot to check.
+     * @return {@code true} when the slot is part of the viewed inventories.
      */
-    boolean hasViewers();
+    boolean isViewedSlot(Slot slot);
 
     /**
-     * Shows this Inventory to the given viewer.
+     * Returns the list of viewed inventories.
+     * <p>This is usually at least the inventory a player opened and the players inventory.</p>
+     * <p>It is not necessary, that all slots of the viewed inventories are visible or interactable with.</p>
      *
-     * @param viewer The viewer to show this inventory to
-     * @throws IllegalArgumentException if a {@link PluginContainer} is not the root of the cause
+     * @return the list of viewed inventories.
      */
-    void open(Player viewer) throws IllegalArgumentException;
+    List<Inventory> getViewed();
```

Comment:
Inventories are composite by nature, why is this needed when you can just query what you want?